### ISSUE TITLE
Can't open select boxes for custom entities that have relations on other custom entities

### ISSUE THREAD

**Issue by Qonstrukt** [cofoundry-cms/cofoundry#222]
I'm not sure this is a new issue, but I've noticed it now I'm trying to implement blog posts with a selectable author and category.

When editing an existing blog post it's impossible to open the select box to make another choice.
When creating a new blog post it only seems to become impossible when I've selected an image to accompany the blog post. Since it's required that might also be the reason I can't open the select boxes anymore when editing a post.

I've tried in Safari and Chrome, both trigger no JavaScript errors. So I don't really have a clue where to look for answers.

**Comment by HeyJoel** [comment]
Hi, I'm trying to replicate this issue in the [SimpleSite](https://github.com/cofoundry-cms/Cofoundry.Samples.SimpleSite) sample project by adding a `CategoryId` property to the blog post data model:

```csharp
public class BlogPostDataModel : ICustomEntityDataModel
{
    [MaxLength(1000)]
    [Required]
    [Display(Description = "A description for display in search results and in the details page meta description.")]
    [MultiLineText]
    public string ShortDescription { get; set; }

    [Image(MinWidth = 460, MinHeight = 460)]
    [Display(Name = "Thumbnail Image", Description = "Square image that displays against the blog in the listing page.")]
    public int ThumbnailImageAssetId { get; set; }

    [Display(Name = "Categories", Description = "Drag and drop to customize the category ordering.")]
    [CustomEntityCollection(CategoryCustomEntityDefinition.DefinitionCode, IsOrderable = true)]
    public ICollection<int> CategoryIds { get; set; }

    [Display(Name = "Category")]
    [CustomEntity(CategoryCustomEntityDefinition.DefinitionCode)]
    public int CategoryId { get; set; }
}
```

I can't seem to replicate the issue though, is that a simplar property definition to the one you're using?

**Comment by Qonstrukt** [comment]
I'll try and make a good repro this weekend!
Apparently there's something particular that triggers the issue.
Thanks for looking into it!

**Comment by Qonstrukt** [comment]
It was such good weather this weekend that I've spent most of it in my pool. 😉 
But I've checked what's happening under water, and it's doing the XHR request to retrieve custom entities, but it turns up empty.
Could this be caused by the fact that my BlogPost has `AutoPublish => false` and `HasLocale => true`, while the BlogAuthor has `AutoPublish => true` and `HasLocale => false`?

**Comment by HeyJoel** [comment]
Good spot, yes locales would be the likely culprit as that's all functionality that still needs to be worked on. 

What you suggest is the intended functionality, so this is a bug and I imagine it is something to do with some misbehaving auto-locale filtering that is trying to automatically filter your author based on the blog locale, but not taking into account that the author is a non-localized entity.

I'll investigate a more as part of v0.4. Is the work around acceptable for now? v0.4 is a little way off yet.

**Comment by Qonstrukt** [comment]
That's ok, this works for now. 👍

**Comment by HeyJoel** [comment]
Fixed in v0.4 release.
